multimodal/src/autogluon/multimodal/models/huggingface_text.py

```diff
@@ -61,18 +72,37 @@ def __init__(
                     - 'xlm-roberta-base'
         num_classes
             The number of classes. 1 for a regression task.
+        pooling_mode
+            The pooling mode for the Transformer. Can be "cls", or "mean"
+        gradient_checkpointing
+            Whether to enable gradient checkpointing
         """
         super().__init__()
         logger.debug(f"initializing {checkpoint_name}")
         self.checkpoint_name = checkpoint_name
         self.num_classes = num_classes
+
         self.model = AutoModel.from_pretrained(checkpoint_name)
+        if isinstance(self.model, T5PreTrainedModel):
+            self.is_t5 = True
+            # Remove the decoder in T5
+            del self.model.decoder
+        else:
+            self.is_t5 = False
+
+        self.gradient_checkpointing = gradient_checkpointing
+        if gradient_checkpointing:
+            self.model.gradient_checkpointing_enable()
+            if self.is_t5:
+                self.dummy_layer = DummyLayer()
```

Comment:
Is this gradient checkpointing issue unique to the `t5` model? If it also exist in other models, the dummy layer shouldn't condition only on `t5`.